java/flight/src/test/java/org/apache/arrow/flight/TestServerOptions.java

```diff
@@ -54,6 +56,47 @@ public void builderConsumer() throws Exception {
     }
   }
 
+  /**
+   * Make sure that if Flight supplies a default executor to gRPC, then it is closed along with the server.
+   */
+  @Test
+  public void defaultExecutorClosed() throws Exception {
+    try (
+        BufferAllocator a = new RootAllocator(Long.MAX_VALUE);
+        FlightServer server =
+            FlightTestUtil.getStartedServer(
+                (location) -> FlightServer.builder(a, location, new NoOpFlightProducer())
+                    .build()
+            )) {
+      Assert.assertTrue("Resources list is non-empty.", server.resources.size() > 0);
+      // We can't actually look for the executor object, as what's in the resources list is a reference to its
+      // shutdown method.
+    }
```

Comment:
can't you try shutting down here and verify it does shutdown?

Reply:
No, there's no accessible reference to the actual executor to confirm it afterwards.

Reply:
I could plumb one through, but it'd be adding references specifically for testing.

(Sorry for the delay again, October was quite the busy month.)

Reply:
What do you think about making resources less generic and take the reference to ExecutorService in the class instead of list of AutoCloseable?  Then make awaitTermination rely on both the server and the executor (if its not null).  That way you don't need anything specifically visible for testing.

Reply:
That makes sense. I've updated the PR.